apps/builder/app/builder/features/style-panel/sections/transitions/transition-timing.tsx

```diff
@@ -0,0 +1,185 @@
+import { useEffect, useMemo, useState } from "react";
+import type { KeywordValue, TupleValueItem } from "@webstudio-is/css-engine";
+import {
+  Combobox,
+  ComboboxAnchor,
+  ComboboxContent,
+  ComboboxListbox,
+  InputField,
+  Label,
+  NestedInputButton,
+  useCombobox,
+  css,
+  ComboboxLabel,
+  theme,
+  ComboboxListboxItem,
+  ComboboxSeparator,
+} from "@webstudio-is/design-system";
+import {
+  defaultFunctions,
+  easeInFunctions,
+  easeOutFunctions,
+  easeInOutFunctions,
+  timingFunctions,
+  findTimingFunctionFromValue,
+  type TimingFunctions,
+  type EaseInFunction,
+  type EaseOutFunction,
+  type DefaultFunction,
+  type EaseInOutFunction,
+} from "./transition-utils";
+
+type NameAndLabel = { name: TimingFunctions; value: string };
+
+type TransitionTimingProps = {
+  timing: KeywordValue;
+  onTimingSelection: (timing: TupleValueItem) => void;
+};
+
+const comboBoxStyles = css({ zIndex: theme.zIndices[1] });
+
+export const TransitionTiming = ({
+  timing,
+  onTimingSelection,
+}: TransitionTimingProps) => {
+  const [inputValue, setInputValue] = useState(
+    findTimingFunctionFromValue(timing.value) ?? timing.value ?? ""
+  );
+  useEffect(
+    () =>
+      setInputValue(
+        findTimingFunctionFromValue(timing.value) ?? timing.value ?? ""
+      ),
+    [timing.value]
+  );
+
+  const {
+    items,
+    isOpen,
+    getComboboxProps,
+    getToggleButtonProps,
+    getInputProps,
+    getMenuProps,
+    getItemProps,
+  } = useCombobox<NameAndLabel>({
+    items: (Object.keys(timingFunctions) as TimingFunctions[]).map((prop) => ({
+      name: prop,
+      value: timingFunctions[prop],
+    })),
+    value: {
+      name: inputValue as TimingFunctions,
+      value: timingFunctions[inputValue as TimingFunctions] ?? "",
+    },
+    selectedItem: undefined,
+    itemToString: (value) => value?.name ?? "",
+    onInputChange: (value) => setInputValue(value ?? ""),
+    onItemSelect: (prop) => {
+      setInputValue(prop.name);
+      onTimingSelection({ type: "keyword", value: prop.value });
+    },
+  });
+
+  const defaults = useMemo(() => {
```

Comment:
seems like quite  bit of code to filter and the items in each category, I have a feeling this could be rewritten in a more compact way and extracted into utility function  (same module) to make the component easier to read

Reply:
I will extract the filtering alone into a function and then make the changes. Thought of it, but missed it refactoring. Will change and push. And then move to the rest two inputs and we can close the transition section 👍